web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/DatastreamController.java

```diff
@@ -116,21 +124,21 @@ private void recordDownloadEvent(PID pid, String datastream, AccessGroupSet prin
     }
 
     @RequestMapping("/thumb/{pid}")
-    public void getThumbnailSmall(@PathVariable("pid") String pid,
-            @RequestParam(value = "size", defaultValue = "small") String size, HttpServletRequest request,
-            HttpServletResponse response) throws IOException {
+    public ResponseEntity<InputStreamResource> getThumbnailSmall(@PathVariable("pid") String pid,
+            @RequestParam(value = "size", defaultValue = SMALL) String size) {
 
-        getThumbnail(pid, size, request, response);
+        return getThumbnail(pid, size);
     }
 
     @RequestMapping("/thumb/{pid}/{size}")
-    public void getThumbnail(@PathVariable("pid") String pidString,
-            @PathVariable("size") String size, HttpServletRequest request,
-            HttpServletResponse response) throws IOException {
+    public ResponseEntity<InputStreamResource> getThumbnail(@PathVariable("pid") String pidString,
+                                                            @PathVariable("size") String size) {
 
         PID pid = PIDs.get(pidString);
         AccessGroupSet principals = getAgentPrincipals().getPrincipals();
-        String thumbName = "thumbnail_" + size.toLowerCase().trim();
+        if (!size.equals(SMALL) && !size.equals(LARGE)) {
```

Comment:
might be overkill given that there are only two sizes, but we could make a constant mapping for the sizes, like
```
private static final Map<String, Integer> THUMB_SIZE_MAP = Map.of(SMALL, 64, LARGE, 128);
```
and then use the map to validate size name and to retrieve the pixel size. 

Also, we should convert the size to lowercase since we were previously.